YoCode/GitCheck.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Diagnostics;
+using System.IO;
+
+namespace YoCode
+{
+    public class GitCheck
+    {
+        public string GIT_INSTALL_DIRECTORY { get; set; } = @"C:\Program Files\Git";
+        //public string REPOSITORY_PATH = @"..\..\..\..\..\YoCode";
+        public string REPOSITORY_PATH = @"C:\Users\ukmzil\source\repos\Tests Sent by People\Real\simon-jones";
+
+        string output { get; set; }
+        string lastAuthor { get; set; }
+        bool result { get; set; }
+
+
+        public List<string> hostDomains = new List<string>();
+
+
+        public bool ExecuteTheCheck()
+        {
+            var p = new Process();
+            return ExecuteTheCheck(p);
+        }
+
+        public bool ExecuteTheCheck(Process p)
+        {
+            p.StartInfo = SetProcessStartInfo(REPOSITORY_PATH);
+            p.Start();
+
+            output = p.StandardOutput.ReadToEnd();
+            lastAuthor = getLastAuthor(output);
+            return GitHasBeenUsed(lastAuthor,hostDomains);
+        }
+
+        public ProcessStartInfo SetProcessStartInfo(String PATH)
+        {
+            var psi = new ProcessStartInfo();
+            psi.CreateNoWindow = true;
+            psi.RedirectStandardError = true;
+            psi.RedirectStandardOutput = true;
+            psi.WorkingDirectory = PATH;
+            psi.FileName = GIT_INSTALL_DIRECTORY + @"\bin\git.exe";
+            psi.Arguments = "log ";
+
+            return psi;
+
+        }
+
+        // get name and email address of the last author
+        public string getLastAuthor(string output)
+        {
+            var sr = new StringReader(output);
+            string line;
+            while ((line = sr.ReadLine()) != null)
+            {
+                if (ContainsAll(line,getKeyWords() ) )
+                {
+                    return line;
+                }
+            }
+            return "";
+        }
+
+
+        public bool GitHasBeenUsed(string lastAuthor, List<string> hostDomains)
+        {
+            if (ContainsAny(lastAuthor,getHostDomains()) || string.IsNullOrEmpty(lastAuthor) )
+            {
+                result = false;
+                return false;
+            }
```

Comment:
Setting `result` and also returning the value seems a bit odd. It might be better to just return the value and then the calling code (line 36) can set result. That would remove a bit of duplication. Like this:
```
result = GitHasBeenUsed(lastAuthor,hostDomains);
return result;
```

Reply:
Sorry, I forgot to delete this, found a way around it